chain-indexer/src/infra/storage/postgres.rs

```diff
@@ -271,7 +280,27 @@ async fn save_transactions(
         .await?;
 
     for (transaction, transaction_id) in transactions.iter().zip(transaction_ids.iter()) {
-        save_contract_actions(&transaction.contract_actions, *transaction_id, tx).await?;
+        let contract_action_ids =
+            save_contract_actions(&transaction.contract_actions, *transaction_id, tx).await?;
+
+        let contract_balances: Vec<(i64, ExtractedBalance)> = transaction
+            .contract_actions
+            .iter()
+            .zip(contract_action_ids.iter())
+            .filter_map(|(action, &action_id)| {
+                ExtractedBalance::extract(&action.state, network_id)
+                    .map(|balances| {
+                        balances
+                            .into_iter()
+                            .map(|balance| (action_id, balance))
+                            .collect::<Vec<_>>()
```

Comment:
This whole expression seems to allocate too many Vecs: First `ExtractedBalance::extract`, then here and then below another collect. We should use try to use Iterators only.